File: python/4th_week/contohsoal2/PenjualBahanBangunan.py
from TokoBangunan import TokoBangunan

class PenjualBahanBangunan:

    
    def __init__(self, noKtp = "", nama = "", banyakToko = 0):
        self.noKtp = noKtp
        self.nama = nama
        self.banyakToko = banyakToko
        self.toko = [TokoBangunan() for i in range(banyakToko)]

    # set and get method
    def setNoKtp(self, paramNoKtp):
        self.noKtp = paramNoKtp
        
    def setNama(self, paramNama):
        self.nama = paramNama
        
    def setBanyakToko(self, paramBanyakToko):
        self.banyakToko = paramBanyakToko
        
    def setToko(self, paramToko, nomorToko):
        self.toko[nomorToko] = paramToko

    def getNoKtp(self):
        return self.noKtp
        
    def getNama(self):
        return self.nama
        
    def getBanyakToko(self):
        return self.banyakToko
        
    def getToko(self):
        return self.toko
    
    # method print semua data
    def cetakPenjual(self):
        print("------------------------------")
        print(f"Nama Penjual       : {self.nama}")
        print(f"No KTP Penjual     : {self.noKtp}")
        print(f"Toko Yang Dimiliki :")
        for i in range(self.banyakToko):
            print(i+1)
            print(f"- Kode Toko   : {self.toko[i].getKodeToko()}")
            print(f"- Nama Toko   : {self.toko[i].getNamaToko()}")
            print(f"- Alamat Toko : {self.toko[i].getAlamatToko()}")
        
    # method print data penjual dan toko tertentu
    def cetakPenjualToko(self, i):
        print("------------------------------")
        print(f"Nama Penjual       : {self.nama}")
        print(f"No KTP Penjual     : {self.noKtp}")
        print(f"Toko Yang Dimiliki :")
        
        print(i+1)
        print(f"- Kode Toko   : {self.toko[i].getKodeToko()}")
        print(f"- Nama Toko   : {self.toko[i].getNamaToko()}")
        print(f"- Alamat Toko : {self.toko[i].getAlamatToko()}")

        
        

File: cpp/4th_week/contohsoal2/PenjualBahanBangunan.cpp
#include "TokoBangunan.cpp"

class PenjualBahanBangunan
{
private:
    /* data */
    string noKtp, nama;
    int banyakToko;
    static TokoBangunan toko[10];
public:
    PenjualBahanBangunan() {}
    PenjualBahanBangunan(string noKtp, string nama, int banyakToko) {
        this->noKtp = noKtp;
        this->nama = nama;
        this->banyakToko = banyakToko;
        toko[banyakToko];
    }

    //  get and set noKtp
    string getnoKtp() {
        return noKtp;
    }

    void setnoKtp(string noKtp) {
        this->noKtp = noKtp;
    }

//    get and set nama
    string getNama() {
        return nama;
    }

    void setNama(string nama) {
        nama = nama;
    }

//    get and set Banyak Toko
    int getBanyakToko() {
        return banyakToko;
    }

    void setBanyakToko(int banyakToko) {
        this->banyakToko = banyakToko;
    }

//   get and set Toko
    TokoBangunan getToko() {
        return this->toko[banyakToko];
    }

    void setToko(TokoBangunan toko, int nomorToko) {
        this->toko[nomorToko] = toko;
    }

//  method
    void cetakPenjual(){
        cout << "------------------------" << endl;
        cout << "Nama Penjual       : " << this->nama << endl;
        cout << "No. KTP Penjual    : " << this->noKtp << endl;
        cout << "Toko Yang Dimiliki : " << endl;
        for (int i = 0; i < banyakToko; i++)
        {
            cout << i+1 << endl;
            cout << "- Kode Toko    : " << toko[i].getKodeToko() << endl;
            cout << "- Nama Toko    : " << toko[i].getNamaToko() << endl;
            cout << "- Alamat Toko  : " << toko[i].getAlamatToko() << endl;
        }
    }

    //method menampilkan penjual dan toko tertentu
    void cetakPenjualToko(int i){
        cout << "------------------------" << endl;
        cout << "Nama Penjual       : " << this->nama << endl;
        cout << "No. KTP Penjual    : " << this->noKtp << endl;
        cout << "Toko Yang Dimiliki : " << endl;
        
        cout << i+1 << endl;
        cout << "- Kode Toko    : " << toko[i].getKodeToko() << endl;
        cout << "- Nama Toko    : " << toko[i].getNamaToko() << endl;
        cout << "- Alamat Toko  : " << toko[i].getAlamatToko() << endl;
        
    }
    ~PenjualBahanBangunan() {}
};